src/main/java/com/example/onlinebookstore/repository/BookRepositoryImpl.java

```diff
@@ -0,0 +1,48 @@
+package com.example.onlinebookstore.repository;
+
+import com.example.onlinebookstore.model.Book;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.EntityManagerFactory;
+import jakarta.persistence.EntityTransaction;
+import jakarta.persistence.TypedQuery;
+import java.util.List;
+import org.hibernate.HibernateException;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl extends AbstractRepository implements BookRepository {
+
+    protected BookRepositoryImpl(EntityManagerFactory entityManagerFactory) {
```

Comment:
this shouldn't be `protected`

honestly I'm not sure how Spring resolves this (probably via reflection `setAccessible` method), as protected constructor is inaccessible for it

it is better to use public constructor